nilearn/__init__.py

```diff
@@ -51,6 +51,18 @@ def _py2_deprecation_warning():
                       stacklevel=3,
                       )
 
+def _py34_deprecation_warning():
+    warnings.simplefilter('once')
```

Comment:
Hey @kchawla-pi 

This line (as well as the same line in `_py2_deprecation_warning`) permanently change the `warnings` filters for all warning types, from all modules/files/lines everywhere.

In my scenario, I explicitly configure warnings, displaying deprecation warnings by default but masking other warnings, but after `nilearn` is imported, my configuration gets overridden by this statement.

How about nesting this block under a `with warnings.catch_warnings():`? That would allow you to temporarily display deprecation warnings, but after doing so, you'd be reverting the warning filters to their prior settings. Do you think that would be a good idea?

Reply:
Thanks @dimrozakis I will look into this! If this is filtering all warnings, that is definitely unintended & undesirable behavior.
Would you send me a code sample which alerted you to this behavior? This was missed this and would like a more robust test for this.

Reply:
Hey @kchawla-pi 

The following snippet:
```python
#!/usr/bin/env python2
import warnings
warnings.warn("Please ignore this warning: 1")  # Will be printed.
warnings.filterwarnings("ignore", message="Please ignore this warning")
warnings.warn("Please ignore this warning: 2")  # Will not be printed.
import nilearn
warnings.warn("Please ignore this warning: 3")  # Should not be printed but is.
```

produces this output:
```
./a.py:3: UserWarning: Please ignore this warning: 1
  warnings.warn("Please ignore this warning: 1")
./a.py:6: DeprecationWarning: Python2 support is deprecated and will be removed in a future release. Consider switching to Python3.
  import nilearn
./a.py:7: UserWarning: Please ignore this warning: 3
  warnings.warn("Please ignore this warning: 3")
```

Warning 1 gets printed, but after adding an ignore filter for warnings matching a particular string, warning 2 isn't printed. Then nilearn gets imported, which prints its deprecation warning. After that, warning 3 gets printed because the filter that ignored warning 2 has been overridden by the filter configured in nilearn. The ignore filter I had configured still exists in `warnings.filters`. But the filter added by nilearn matches first so the warning gets printed.

Hope the above help to illustrate the problem.

Reply:
An alternative to using the `warnings.catch_warnings` context manager would be to configure a global warnings filter that is specific enough so that it doesn't mess with other warnings and filters. Filters can be configured by category, message regex, module regex, even line number iirc.

Reply:
Thanks, this is very helpful. I'll look into this.